docs/gallery/domain/plot_icephys.py

```diff
@@ -457,6 +458,38 @@
     category="electrodes",
 )
 
+#####################################################################
+# One predefined subcategory column is the stimulus template column in the stimuli table. This column can be used to
+# store stimulus template waveforms that were generated by the recording software. Similar to the stimulus and response
+# columns, we can specify a relevant time range.
+
+nwbfile.intracellular_recordings.add_column(
+    name="stimulus_template",
+    data=[TimeSeriesReference(0, 5, stimulus),  # (start_index, index_count, stimulus_template)
+          TimeSeriesReference(1, 3, stimulus),
+          TimeSeriesReference.empty(stimulus)],  # if there was no data for that recording, use an empty reference
+    description="Column storing the reference to the stimulus template for the recording (rows).",
+    category="stimuli",
+    col_cls=TimeSeriesReferenceVectorData
+)
+
+# we can also add stimulus template data as follows
+rowindex = nwbfile.add_intracellular_recording(
+    electrode=electrode,
+    stimulus=stimulus,
+    stimulus_template=stimulus,  # the full time range of the stimulus template will be used unless specified
```

Comment:
We should not use the same TimeSeries for the `stimulus` and `stimulus_template`. I would suggest we create a new TimeSeries for the stimulus template and add it in the proper location for templates. Looking at the schema, the presented stimulus should I believe go into the `stimulus/presentation` group and the template in `stimulus/templates` group. Similarly, above we should use the template not the actual stimulus when adding data to the `stimulus_template` column.